include/libmatrix-client/HTTP.h

```diff
@@ -0,0 +1,42 @@
+#ifndef HTTP_H
+#define HTTP_H
+
+#include <string>
+#include <map>
+
+namespace LibMatrix {
+	struct CaseInsensitiveCompare {
+		bool operator()(const std::string& a, const std::string& b) const noexcept;
+	};
+
+	using Headers = std::map<std::string, std::string, CaseInsensitiveCompare>;
+
+	enum class HTTPMethod {
+		CONNECT, DEL, GET, HEAD, OPTIONS,
+		POST, PUT, TRACE, PATCH
+	};
+
+	class Response {
+	public:
+		const std::string data;
+		const int status;
```

Comment:
I would provide an `enum`, which would convey the meaning of the status better than a regular integer.

We can return the status code and allow comparisons to that `enum`.